vizro-core/src/vizro/models/_components/table.py

```diff
@@ -51,12 +53,35 @@ def __getitem__(self, arg_name: str):
             return self.type
         return self.figure[arg_name]
 
+    def _build_underlying_table_object(self):
+        data = data_manager._get_component_data(self.id)  # type: ignore
+        kwargs = self.figure._arguments.copy()
+        kwargs.pop("data_frame", None)
+        return self.figure._function(data_frame=data, **kwargs)
+
+    @_log_call
+    def pre_build(self):
+        if self.actions:
+            # The underlying table object is pre-built, so we can fetch its ID.
+            underlying_table_object = self._build_underlying_table_object()
+
+            # Underlying table object has to have "id" defined if it triggers actions chain.
+            if not hasattr(underlying_table_object, "id"):
+                raise ValueError(
+                    "Underlying table object has no attribute 'id'. To perform specified actions, a valid 'id' to the"
+                    " Underlying table object has to be provided."
+                )
+
+            self._underlying_table_id = underlying_table_object.id
+
     @_log_call
     def build(self):
         return html.Div(
             [
                 html.H3(self.title, className="table-title") if self.title else None,
-                html.Div(dash_table.DataTable(), id=self.id),
+                html.Div(
+                    dash_table.DataTable(**({"id": self._underlying_table_id} if self.actions else {})), id=self.id
+                ),
```

Comment:
After we support AgGrid and others, we will also need to take care of the empty object we create here.

Reply:
Do we really need the if clause? In the end why not always attach ID of the pre-built table?

Reply:
Depending on what's slicker in code, I would probably just return an empty datatable even for the ag-grid.

Visually we will aim to style them very similar, so the empty figure shouldn't look different when you use the AgGrid or the DataTable

Reply:
> Do we really need the if clause? In the end why not always attach ID of the pre-built table?

Underlying table ID isn't always provided and it isn't always necessary. 
Underlying table ID is necessary only if `vm.Table` has `actions` defined (which means that underlying table triggers action/-s).

The reason why we need `id` defined in the empty `datatable` when the build method is invoked is because this `datatable` is input into `_on_page_load` action (so it has to exist in UI elements). We can potentially change this behaviour and say that `filter_interaction` doesn't need to be the input of the `_on_page_load` action, but we would probably need to discuss it. 

> I would probably just return an empty datatable even for the ag-grid

Since `filter_interaction` input will be different ("active_cell" and "derived_viewport_data" for `datatable` and "cellClicked" for `AgGrid`) it means that probably we would need to take care of what empty component is created as well. I guess it will happen that the Dash error will be raised because somewhere this `datatable` uses `"cellClicked"` property which is not available. I'm not sure so let's discuss it once we add AgGrid and others if you agree.

Since the `filter_interaction` input will be different ("active_cell" and "derived_viewport_data" for `datatable` and "cellClicked" for `AgGrid`), this means we should probably also care about which empty component is created. I suspect that the Dash error will occur because in `on_page_load` action this `datatable` is using a `"cellClicked"` property that is not available. I'm not sure so let's discuss when we add AgGrid and others if you agree.

Reply:
Yes, that's fine @petar-qb 👍  I wouldn't worry too much of making everything compatible with the AgGrid in this PR. This can be done in a separate PR. The public API should just not change, which I think we already ensured in Max previous PR. So happy to move all considerations regarding the AgGrid which are more complex to implement right now in a separate ticket